token/confidential-transfer/proof-generation/src/mint.rs

```diff
@@ -77,13 +79,34 @@ pub fn mint_split_proof_data(
         )
         .ok_or(TokenProofGenerationError::IllegalAmountBitLength)?;
 
-    // decrypt the current supply
-    let current_supply = current_decryptable_supply
-        .decrypt(supply_aes_key)
-        .ok_or(TokenProofGenerationError::IllegalAmountBitLength)?;
+    // fresh mints are initialized with a zeroed decryptable_supply
+    // TODO: don't clone here once AeCiphertext implement Copy in the zk-sdk
+    let pod_decryptable_supply: PodAeCiphertext = current_decryptable_supply.clone().into();
+    let current_decyptable_supply = if pod_decryptable_supply != PodAeCiphertext::default() {
+        // decrypt the current supply
+        current_decryptable_supply
+            .decrypt(supply_aes_key)
+            .ok_or(TokenProofGenerationError::IllegalAmountBitLength)?
+    } else {
+        0
+    };
+
+    // get the difference between the supply ciphertext and the decryptable supply
+    // explanation see https://github.com/solana-labs/solana-program-library/pull/6881#issuecomment-2385579058
+    let decryptable_supply_ciphertext = supply_elgamal_keypair
+        .pubkey()
+        .encrypt(current_decyptable_supply);
+    #[allow(clippy::arithmetic_side_effects)]
+    let ct_decryptable_to_current_diff = decryptable_supply_ciphertext - current_supply_ciphertext;
```

Comment:
Nit: let's avoid abbreviations here. I am thinking that `supply_delta_ciphertext` might suffice for the variable name?